src/main/java/control/DetailViewController.java

```diff
@@ -497,55 +474,104 @@ private void activeBlockChangedDirector() {
             .setText(detailView.formatDouble(shotBlock.getPaddingBefore()));
         ((DirectorDetailView) detailView).getPaddingAfterField()
             .setText(detailView.formatDouble(shotBlock.getPaddingAfter()));
-        activeBlock = shotBlock;
-        initInstrumentsDropdown(shotBlock);
+        activeDirectorBlock = shotBlock;
         detailView.setVisible();
         // Re-init the detail view with new data
         manager.getRootPane().getRootHeaderArea().reInitHeaderBar(detailView);
         this.reInitForDirectorBlock();
     }
 
     /**
-     * Initialize drop down menu.
+     * Initialize drop down menu for director camera selection.
      */
-    private void initDropDown() {
+    private void initDirectorCameraDropdown() {
         StyledMenuButton cameraButtons = ((DirectorDetailView) detailView).getSelectCamerasButton();
         cameraButtons.setBorderColor(TweakingHelper.getColor(0));
         cameraButtons.setFillColor(TweakingHelper.getBackgroundColor());
-        activeBlockBoxes = new ArrayList<>();
+        activeCameraBoxes = new ArrayList<>();
+
+        cameraButtons.showingProperty().addListener(createCameraDropdownListener(cameraButtons));
+    }
+
+    /**
+     * Initialize drop down menu for camera instrument selection.
+     */
+    private void initCameraInstrumentDropdown() {
+        StyledMenuButton instrumentsButtons =
+                ((DetailView) detailView).getSelectInstrumentsButton();
+        instrumentsButtons.setBorderColor(TweakingHelper.getColor(0));
+        instrumentsButtons.setFillColor(TweakingHelper.getBackgroundColor());
+        activeInstrumentBoxes = new ArrayList<>();
+
+        instrumentsButtons.showingProperty().addListener(
+                createInstrumentsDropdownListener(instrumentsButtons));
+    }
+
+    /**
+     * Creates ChangeListener for the Instruments Dropdown checkboxes.
+     * @param buttons the dropdown with checkboxes.
+     * @return the ChangeListener.
+     */
+    private ChangeListener<Boolean> createInstrumentsDropdownListener(StyledMenuButton buttons) {
+        return new ChangeListener<Boolean>() {
+            @Override
+            public void changed(ObservableValue<? extends Boolean> observable,
```

Comment:
If you subtract the functionality of changed into a seperate function it will be more readable and more easily testable